src/engine/LocalVocab.cpp

```diff
@@ -8,6 +8,27 @@
 #include "global/Id.h"
 #include "global/ValueId.h"
 
+// _____________________________________________________________________________
+LocalVocab LocalVocab::clone() {
+  LocalVocab localVocabClone;
+  // First, make a deep copy of the `absl::node_hash_map` holding the actual
+  // map of strings to indexes.
+  localVocabClone.wordsToIdsMap_ = this->wordsToIdsMap_;
+  // The next free index should be the same.
+  localVocabClone.nextFreeIndex_ = this->nextFreeIndex_;
+  // The map from local ids to strings stores pointers to strings, so we cannot
+  // just copy these, but make sure to use the pointers from the keys of the new
+  // map, and not of the original map.
+  localVocabClone.idsToWordsMap_.reserve(this->size());
+  for (const std::string* wordPtr : this->idsToWordsMap_) {
+    auto it = localVocabClone.wordsToIdsMap_.find(*wordPtr);
+    AD_CHECK(it != localVocabClone.wordsToIdsMap_.end());
+    localVocabClone.idsToWordsMap_.push_back(std::addressof(it->first));
+  }
```

Comment:
It should be much more efficient to do this the other way round: iterate over the HashMap and insert into the vector at the correct position.
If you don't want the random access:
First create a vector of `pair<string*, index>`, then sort it, and then extract only the pointers.
In C++23 you can even use a `zip_view` of two separate vectors, avoiding the copy at the end (make this a *TODO*).